include/xten/Dialect/XTenNN/Interfaces/EnclaveOpInterface.td

```diff
@@ -37,6 +37,19 @@ def XTenNN_EnclaveOp : OpInterface<"EnclaveOp"> {
             /*methodBody=*/"",
             /*defaultImplementation=*/[{ return $_op.getOperands(); }]
         >,
+        InterfaceMethod<
+            /*desc=*/[{Gets the enclave body block, if present.}],
+            /*retTy=*/"::mlir::Block *",
+            /*methodName=*/"getOptionalEnclaveBody",
+            /*args=*/(ins),
+            /*methodBody=*/"",
+            /*defaultImplementation=*/[{
+                auto &region = $_op->getRegion(0);
```

Comment:
Shouldn't we use `getRegions()` and if it's not empty, we return the front block instead of accessing directly by index?

Reply:
I took the same approach as `func.func` so the region is created regardless but the body is either present of absent.